common/djangoapps/student/tasks.py

```diff
@@ -0,0 +1,104 @@
+"""
+Celery task for course enrollment email
+"""
+import logging
+
+from celery import shared_task
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from opaque_keys.edx.keys import CourseKey
+
+from common.djangoapps.course_modes.models import CourseMode
+from common.djangoapps.student.helpers import get_course_dates_for_email, get_instructors
+from lms.djangoapps.utils import get_braze_client
+from openedx.core.djangoapps.catalog.utils import (
+    get_course_uuid_for_course,
+    get_owners_for_course,
+    get_course_run_details
+)
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.features.course_experience import ENABLE_COURSE_GOALS
+
+User = get_user_model()
+log = logging.getLogger(__name__)
+
+MAX_RETRIES = 1
+COUNTDOWN = 5
+
+
+@shared_task(bind=True, ignore_result=True)
+def send_course_enrollment_email(self, user_id, course_id, course_title, short_description, pacing_type):
+    """
+    Send course enrollment email using Braze API
+    """
+
+    course_run_fields = [
+        'key', 'title', 'short_description', 'marketing_url', 'pacing_type', 'min_effort',
+        'max_effort', 'weeks_to_complete', 'enrollment_count', 'image', 'staff',
+    ]
+    canvas_entry_properties = {
+        'course_title': course_title,
+        'short_description': short_description,
+        'pacing_type': pacing_type,
+        'course_run_key': course_id,
+        'course_price': CourseMode.min_course_price_for_currency(course_id=course_id, currency='USD'),
+        'lms_base_url': configuration_helpers.get_value('LMS_ROOT_URL', settings.LMS_ROOT_URL),
+        'learning_base_url': configuration_helpers.get_value('LEARNING_MICROFRONTEND_URL',
+                                                             settings.LEARNING_MICROFRONTEND_URL),
+    }
+
+    try:
+        user = User.objects.get(id=user_id)
+        course_key = CourseKey.from_string(course_id)
+        course_date_blocks = get_course_dates_for_email(user, course_key, request=None)
+    except Exception:  # pylint: disable=broad-except
+        pass
+    else:
```

Comment:
`else` will only run if the code block has no error. We want to run this for all cases.

```suggestion
    finally:
```